packages/app-project/pages/_middleware.js

```diff
@@ -0,0 +1,16 @@
+import { NextResponse } from 'next/server'
+
+export function middleware(req, event) {
+  const { pathname, searchParams } = req.nextUrl
+  if (searchParams.has('language')) {
+    const locale = searchParams.get('language')
+    const url = req.nextUrl.clone()
+    try {
+      url.locale = locale
```

Comment:
Somewhere around here, the old language query param should be deleted from the new URL. NextJS uses the standard [URLSearchParams API](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) so the syntax is something like:

```js
url.searchParams.delete('language')
```